src/xrpld/app/tx/detail/DepositPreauth.cpp

```diff
@@ -90,14 +155,38 @@ DepositPreauth::preclaim(PreclaimContext const& ctx)
 
         // Verify that the Preauth entry they asked to add is not already
         // in the ledger.
-        if (ctx.view.exists(keylet::depositPreauth(ctx.tx[sfAccount], auth)))
+        if (ctx.view.exists(keylet::depositPreauth(account, auth)))
             return tecDUPLICATE;
     }
-    else
+    else if (ctx.tx.isFieldPresent(sfUnauthorize))
     {
         // Verify that the Preauth entry they asked to remove is in the ledger.
-        AccountID const unauth{ctx.tx[sfUnauthorize]};
-        if (!ctx.view.exists(keylet::depositPreauth(ctx.tx[sfAccount], unauth)))
+        if (!ctx.view.exists(
+                keylet::depositPreauth(account, ctx.tx[sfUnauthorize])))
+            return tecNO_ENTRY;
+    }
+    else if (ctx.tx.isFieldPresent(sfAuthorizeCredentials))
+    {
+        STArray const& authCred(ctx.tx.getFieldArray(sfAuthorizeCredentials));
+        for (auto const& o : authCred)
```

Comment:
Can make this slightly more efficient and populate `std::set`. Then don't need to call `makeSorted` and iterate again.

Reply:
fixed